src/pybind/mgr/volumes/fs/operations/volume.py

```diff
@@ -84,15 +84,23 @@ def connect(self):
             log.debug("CephFS mounting...")
             self.fs.mount(filesystem_name=self.fs_name.encode('utf-8'))
             log.debug("Connection to cephfs '{0}' complete".format(self.fs_name))
-            self.mgr._ceph_register_client(self.fs.get_addrs())
+            r = self.mgr._ceph_register_client(self.fs.get_addrs())
```

Comment:
@batrick commit 969b6604daf6d5bfbf9ee1b9f8a0b753f979bb41 adds a guard in `_ceph_register_client()` to deny registering a client when manager waits for ack from mon for the last sent beacon. between handling this failure and calling disconnect, the ack could arrive and the manager would either shutdown (`exit(0)`) or respawn (active -> standby). For the shutdown case its not a problem since sending the last beacon (and waiting for an ack) is executed from singal handler context, but, when respawning, manager modules could still be executing when manager waits for the last beacon ack.

Reply:
There is an issue even when shutting down from a signal handler -- a module could instantiate a cephfs connection (init, mount) and before registering the handles addrs, a termination signal would never send the just instantiated client addrs resulting in the same problem. As a workaround/fix, the whole fs init/mount and register should happen when signals are blocked.

For the respawn case (comment above) -- mgr could send a SIGTERM to iself and we could `execv()` from the signal handler (`execv` is a async-singnal-safe syscall).